app/models/maintenance_stats/queries/query_utils.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module MaintenanceStats
+  module Queries
+    class QueryError < StandardError; end
+
+    class QueryUtils
+      def self.check_for_graphql_errors(result, repository_name)
+        if result.data.nil?
+          StructuredLog.capture(
+            "GITHUB_STAT_QUERY_ERROR",
+            {
+              repository_name: repository_name,
+              error_messages: ["no data found in result"],
+            }
+          )
+          raise QueryError, "No data found in result"
+        end
+
+        if result.errors.any?
+          error_messages = result.errors.values.flatten
+          StructuredLog.capture(
+            "GITHUB_STAT_QUERY_ERROR",
+            {
+              repository_name: repository_name,
+              error_messages: error_messages,
+            }
+          )
+          raise QueryError, "Error exists in query: #{error_messages}"
+        end
+
+        if result.data.errors.any?
```

Comment:
just to check, have you run into any cases where we get a `data.errors` key back?

Reply:
This [spec](https://github.com/librariesio/libraries.io/blob/270fe79f915321a672b8f4a49201a8562793483c/spec/models/repository_spec.rb#L256-L274) is responding with errors in `data.errors`. I'm going off of what was already here as I don't remember many of the details of how this all works. The GraphQL client page has a [small example](https://github.com/github/graphql-client/blob/master/guides/handling-errors.md) on handling different errors. I think `result.data.errors` would be the most common.